mozilla/fxa
# feat(settings): Signin Recovery Code page in React

## Because

* We are converting backbone routes to React, and this is the last route to be converted to enable React sign in

## This pull request

* Add signin_recovery_code to routes that can be served with React
* Create container component for SigninRecoveryCode page and add logic to hook up the page
* Add unit tests and update storybook
* Fix error with react import for SigninTotpCode and SigninTokenCode

## Issue that this pull request solves

Closes: #FXA-9082

## Checklist

_Put an `x` in the boxes that apply_


## Screenshots (Optional)

Enabled page: 
![image](https://github.com/mozilla/fxa/assets/22231637/0eab9f37-7427-4e87-a039-1cb6bad59733)

## Other information (Optional)

How to test:

- Create a new account and set up Two-factor authentication from /settings (make sure to save backup codes for later!)
- Sign out
- Set `forceGlobally`/`SIGNIN_CONFIRMATION_FORCE_GLOBALLY` to `true` to force the 2FA verification on signin (in auth-server config)
- Sign in with react experiment: http://localhost:3030/?forceExperiment=generalizedReactApp&forceExperimentGroup=react
- When prompted for 2FA, click on "Trouble entering code?" to reach the `signin_recovery_code` page
- Verify that previously saved backup code can be used to login
- Also test with OAuth (123Done) and experiment params added, it should navigate back to the RP with a lock icon to indicate the session is TOTP verified
